typescript/src/util.ts

```diff
@@ -105,3 +105,11 @@ String.prototype.toSnakeCase = function(this: string): string {
   }
   return words.join('_');
 };
+
+Array.prototype.camelCaseBeforeDot = function(this: string[]): string {
+  if (this.length <= 1) {
+    return this.toString().toCamelCase();
+  }
+  const res = this.slice(0, -1).map(w => w.toCamelCase());
+  return res.join('!.') + '!.' + this[this.length - 1];
```

Comment:
Wait. `!.` is a TypeScript specific, we must not have it in code :)  There should be a `for` loop in the templates doing that.

Reply:
Alternatively, if you want to leave it here, pass a string to join as a parameter:
```ts
Array.prototype.camelCaseBeforeDot = function(this: string[], joiner: string): string { ... }
```

then you will call it from a template:
```ts
arr.camelCaseBeforeDot("!.")
```

and it will be fine since the function won't have any TypeScript specific code.